Zend/zend_compile.c

```diff
@@ -5302,6 +5341,23 @@ static zend_type zend_compile_typename(zend_ast *ast, zend_bool force_allow_null
 				zend_string_addref(class_name);
 			}
 
+			if (ast->attr == ZEND_NAME_NOT_FQ
+					&& zend_is_confusable_type(orig_name, &correct_name)
+					&& zend_is_not_imported(orig_name)) {
+				if (correct_name) {
+					zend_error(E_COMPILE_WARNING,
+						"\"%s\" will be interpreted as a class type. Did you mean \"%s\"? "
+						"Use qualified name or \"use\" to suppress this warning",
```

Comment:
"Use" twice in the sentence is hard for me to read. I'd suggest:

> Fully qualify the class name or "use" the class to suppress this warning

Reply:
New phrasings are (depending on whether namespaced or not):

> Write "\boolean" to suppress this warning

> Write "\Foo\boolean" or import the class with "use" to suppress this warning

Sounds fine?